### PR TITLE
Add API Metrics Support to UrlScan

### PR DESCRIPTION
**Pull request by amshamah419**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Description
Added Ratelimit, retries on quota errors, and API Metrics reporting

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details: Moved to Python 3. Should not break BC
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by amshamah419** [comment]
@yuvalbenshalom Can we merge this? I think it'll require a force merge due to the change from py2 to py3.

FYI @bakatzir

**Comment by bakatzir** [comment]
Are we fixing https://github.com/demisto/content/blob/master/Packs/Base/Scripts/CommonServerPython/CommonServerPython.py#L7075
 in this PR, or in a separate one?

**Comment by amshamah419** [comment]
Part of me wants to say we should do it in this PR since we are force merging anyways, but since we'd be touching base pack, I'd like it to be separate just in case.

File: Packs/UrlScan/Integrations/UrlScan/UrlScan.py [hunk 1]
```diff
@@ -263,13 +306,13 @@ def format_results(client, uuid):
     # Scan Lists sometimes returns empty
     num_of_attempts = 0
     relationships = []
-    response = urlscan_submit_request(client, uuid)
+    response, metric_results, _ = urlscan_submit_request(client, uuid)
```
Comment (yuvalbenshalom) [inline Packs/UrlScan/Integrations/UrlScan/UrlScan.py:309]: is this being used?

File: Packs/UrlScan/Integrations/UrlScan/UrlScan.py [hunk 2]
```diff
@@ -503,22 +546,42 @@ def get_urlscan_submit_results_polling(client, uuid):
...
+            continue
+        if metrics == ErrorTypes.QUOTA_ERROR:
+            if not is_scheduled_command_retry():
+                execution_metrics.quota_error += 1
+            items_to_schedule.append(url)
+            continue
         uuid = response.get('uuid')
         get_urlscan_submit_results_polling(client, uuid)
+        execution_metrics.success += 1
+    schedule_and_report(command_results=command_results, items_to_schedule=items_to_schedule,
```
Comment (yuvalbenshalom) [inline Packs/UrlScan/Integrations/UrlScan/UrlScan.py:574]: please run a test(not unit test) where we get an quota error and it is bing handled by this successfully.

**Review by yuvalbenshalom — COMMENTED** [review]
see my comments.

**Comment by yuvalbenshalom** [comment]
and also, the urlsacan test playbook is not running. we must have it here.

**Comment by amshamah419** [comment]
> and also, the urlsacan test playbook is not running. we must have it here.

It doesn't run since it's a partner pack. Locally running the PB checks out though

File: Packs/UrlScan/Integrations/UrlScan/UrlScan.py (hunk 1 above)
Comment (amshamah419) [reply Packs/UrlScan/Integrations/UrlScan/UrlScan.py:309]: It gets passed up to the submission function so we know whether or not to schedule the request.

**Review by yuvalbenshalom — APPROVED** [review]

**Pull request merged**